crypto/ec/ecp_nistp224.c

```diff
@@ -75,6 +75,7 @@ typedef uint64_t u64;
  */
 
 typedef uint64_t limb;
+typedef uint64_t limb_aX __attribute((__aligned__(1)));
```

Comment:
This might be reached by other compilers than GCC in theory, do we need to take care of alternative extensions for unusual alignment depending on the compiler?

https://github.com/openssl/openssl/blob/f7201301ef001b70109d7007a37525e233d30907/crypto/ec/ecp_nistz256.c#L40-L46

https://github.com/openssl/openssl/blob/f7201301ef001b70109d7007a37525e233d30907/crypto/ec/ecp_nistz256_table.c#L24-L30

Reply:
Do we have ways of testing this PR with MSVC (maybe we can change appveyor config to trigger a relevant target?) and Sun C compiler (no idea if a 64-bit solaris or sparc9 build satisfies the requirements for `ec_nistp_64_gcc_128`, but in theory one could use the Sun C compiler also on amd64 and it might support 128 bit types) ?

Reply:
MSVC is not GCC.
Unless I am completely mistaken there is no way how this will work with that compiler.
I would not fix theoretical bugs either, since this looks like ordinary C code, it
is still a challenge for the C-compiler to implement 128 bit unsigned integer
arithmetic correctly.  So unless there is an effective way to test the change I would
defer that to a later PR.

Reply:
FYI, I tried the following program on https://godbolt.org
```C
// Type your code here, or load an example.
#ifndef __GNUC__
#error sss
#endif
typedef __uint128_t uint128_t;
typedef unsigned long long limb;
typedef limb limb_aX __attribute((__aligned__(1)));
int square(int num) {
    return sizeof(limb);
}
```
It complies with clang, and icc, so both compilers define `__GNUC__`
and manage at least to ignore that attribute.

Reply:
Great, I imagine we can always do a small fix PR later on in case a non-gcc compiler checking all the requirements, supporting `__uint128_t` but disliking `__aligned__` were to appear!

Thanks for the extra testing!